src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl

```diff
@@ -662,16 +671,15 @@ function update_auxiliary_state_gradient!(
     FT = eltype(Q)
     A = dg.state_auxiliary
 
-    # store ∇ʰu as integrand for w
-    function f!(m::HBModel, Q, A, D, t)
-        @inbounds begin
-            # load -∇ʰu as ∂ᶻw
-            A.w = -D.∇ʰu
-        end
-
-        return nothing
-    end
-    nodal_update_auxiliary_state!(f!, dg, m, Q, t, elems; diffusive = true)
+    update_auxiliary_state!(
+        nodal_update_auxiliary_state!,
+        dg,
+        m,
+        Q,
+        t,
+        elems;
+        diffusive = true,
+    )
```

Comment:
looking good! What needs to be done to drop the first argument and get this on one line? 😃

Reply:
Looks like we can do
```julia
    f! = nodal_update_auxiliary_state!
    update_auxiliary_state!(f!, dg, m, Q, t, elems; diffusive = true)
```

Reply:
can't you just update the `update_auxiliary_state!(dg::DGModel,balance_law::BalanceLaw,state_prognostic,t,elems)` method with the `diffusive` kwarg? we have pretty similar interfaces everywhere.

Reply:
I think everything is in shape.